pkg/apis/eventing/v1alpha1/broker_validation.go

```diff
@@ -54,9 +55,24 @@ func isValidChannelTemplate(dct *eventingduckv1alpha1.ChannelTemplateSpec) *apis
 }
 
 func (b *Broker) CheckImmutableFields(ctx context.Context, og apis.Immutable) *apis.FieldError {
-	// Currently there are no immutable fields. We could make spec.channelTemplate immutable, as
-	// changing it will normally not have the desired effect of changing the Channel inside the
-	// Broker. It would have an effect if the existing Channel was then deleted, the newly created
-	// Channel would use the new spec.channelTemplate.
+	original, ok := og.(*Broker)
```

Comment:
I noticed the Trigger's CheckImmutableFields has this nil check: https://github.com/knative/eventing/blob/d19a8e411a95dccd287bb2082dfd5f17391f5bb0/pkg/apis/eventing/v1alpha1/trigger_validation.go#L108-L110

Do we not need that here?